containers/src/Data/Sequence/Internal.hs

```diff
@@ -212,6 +211,11 @@ import Data.Monoid (Monoid(..))
 import Data.Functor (Functor(..))
 import Utils.Containers.Internal.State (State(..), execState)
 import Data.Foldable (Foldable(foldl, foldl1, foldr, foldr1, foldMap, foldl', foldr'), toList)
+import qualified Data.Foldable as F
+
+#if !(__GLASGOW_HASKELL__ >= 708)
```

Comment:
Intuitively I'd prefer if CPP for `base` features was done with `MIN_VERSION_base`. I find it hard to remember the `base` version numbers though, so I wouldn't mind if we'd keep using `__GLASGOW_HASKELL__`.

Reply:
Definitely should use `MIN_VERSION_base` for this. Is there any reasonable way to avoid CPP for this altogether?

Reply:
Because he use of `Data.List` in the file is on `__GLASGOW_HASKELL__ >= 708`. https://github.com/haskell/containers/blob/master/containers/src/Data/Sequence/Internal.hs#L4388-L4392

I'm not keen to change / audit the CPP usage. `__GLASGOW_HASKELL_ >= 708` is as good as `MIN_VERSION_base` to guard `coerce` usage.

Reply:
To be clear: say how you want this to be changed. I think the only way to make this nice is to drop pre-GHC-7.8 support (i.e. 7.6). But that rather be done in a separate patch going though whole of `containers`.